symbolic-base/src/ZkFold/Protocol/IVC/Internal.hs

```diff
@@ -157,17 +156,19 @@ ivcProve f res witness =
     in
         IVCResult z' acc' ivcProof
 
-ivcVerify :: forall ctx0 ctx1 algo d k a i p c f . IVCAssumptions ctx0 ctx1 algo d k a i p c f
-    => StepFunction a i p
-    -> IVCResult k i c f
-    -> ((Vector k (c f), [f]), (Vector k (c f), c f))
-ivcVerify f res =
+ivcVerify ::
+    forall d k a i p c f . IVCAssumptions d k a i p c f =>
+    Hasher (FieldElement (CircuitContext a)) ->
+    Hasher f ->
```

Comment:
Unfortunately we have to pass two `Hasher`s here which work over two different base fields since there's no known bijection between them, and an invariant expected here is that they compute the same function.

Note that this problem was present before, but was obfuscated by type aliases for constraint sets!

However this can be solved by instead defining `type Hasher = forall f. C f => [f] -> f`, but I don't know which `C` to put here. `Ring` is enough for `mimcHash`, but I guess we would like to use other hashing algorithms as well.